src/data/markdown/docs/30 xk6-browser/01 xk6-browser/01 Browser.md

```diff
@@ -0,0 +1,35 @@
+---
+title: "Browser"
+excerpt: "xk6-browser: Browser Class"
+---
+
+The `Browser` class is the entry point for all your tests, and it is what interacts with the actual web browser via [Chrome DevTools Protocol](https://chromedevtools.github.io/devtools-protocol/) (CDP). It manages:
+- [BrowserContext](/javascript-api/xk6-browser/browsercontext/)s which is where you can set a variety of attributes to control the behavior of pages;
+- and [Page](/javascript-api/xk6-browser/page/)s which is where your rendered site is displayed.
+
+A new Browser instance (hence a new browser process) can be created using the `launch()` method of the `chromium` module from `'k6/x/browser'`.
+
+| Method                                                                                    | Description                                                                                                                                           |
+|-------------------------------------------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------|
+| [browser.close()](/javascript-api/xk6-browser/browser/close)                              | Closes the browser and all of its pages (if any were opened).                                                                                         |
+| [browser.contexts()](/javascript-api/xk6-browser/browser/contexts)                        | Allows you to access all open [BrowserContext](/javascript-api/xk6-browser/browsercontext/)s.                                                        |
+| [browser.isConnected](/javascript-api/xk6-browser/browser/isconnected) <BWIPT id="453"/>  | Indicates whether the [CDP](https://chromedevtools.github.io/devtools-protocol/) connection to the browser process is active or not.                  |
+| [browser.newContext([options])](/javascript-api/xk6-browser/browser/newcontext/) <BWIPT id="455"/> | Creates and returns a new [BrowserContext](/javascript-api/xk6-browser/browsercontext/).                                                             |
+| [browser.newPage([options])](/javascript-api/xk6-browser/browser/newpage)  <BWIPT id="455"/>       | Creates a new [Page](/javascript-api/xk6-browser/page/) in a new [BrowserContext](/javascript-api/xk6-browser/browsercontext/) and returns the page. |
```

Comment:
Yep, good spot. So i've made the change in `newPage`'s doc page in commit 791e18235b75e233a45a489276579bb82f47f520. Let me know if this works or needs more work.

Reply:
Nice!

Reply:
Correct @mdcruz, a `BrowserContext` is created implicitly with `browser.newPage()`, and `browser.newContext()` is not strictly necessary.

The reason we showcase it often is because `BrowserContext`s behave like separate sessions, so if you create two pages with `browser.newPage()`, they will be completely isolated, which has some behavior implications (e.g. cookie sharing). It's also useful if you want to close many pages (tabs) at once, calling `context.close()` will close all pages within that context.

This is not so useful for smaller scripts, but users might want to be aware of that for more complex cases.